test/matlab_experimental/ctTestConstPressureReactor.m

```diff
@@ -0,0 +1,186 @@
+
+            self.r1 = IdealGasReactor(self.gas1);
+            self.r2 = Reactor(self.gas2);
+
+            self.r1.V = 0.2;
+            self.r2.V = 0.2;
+
+            self.resGas.TP = {T0 - 300, P0};
+            self.env = Reservoir(self.resGas);
+
+            U = 300 * arg.addQ;
+
+            if arg.addMdot
+                self.mfc1 = MassFlowController(self.env, self.r1);
+                self.mfc1.massFlowRate = 0.05;
+                self.mfc2 = MassFlowController(self.env, self.r2);
+                self.mfc2.massFlowRate = 0.05;
+            end
+
+            if arg.addSurf
+                self.interface1 = Interface('diamond.yaml', 'diamond_100', ...
+                                            self.gas1, self.solid);
+                self.interface2 = Interface('diamond.yaml', 'diamond_100', ...
+                                            self.gas2, self.solid);
+
+                C = zeros(1, self.interface1.nSpecies);
+                C(1) = 0.3;
+                C(5) = 0.7;
+                C = {C};
+
+                self.surf1 = ReactorSurface(self.interface1, self.r1);
+                self.surf1.area = 0.2;
+                self.interface1.coverages = C;
+                self.surf2 = ReactorSurface(self.interface2, self.r2);
+                self.surf2.area = 0.2;
+                self.interface2.coverages = C;
+            end
+
+            self.net1 = ReactorNet(self.r1);
+            self.net2 = ReactorNet(self.r2);
+            self.net1.maxTimeStep = 0.05;
+            self.net2.maxTimeStep = 0.05;
+            % self.net2.maxErrTestFails = 10;
+        end
+
+        function integrate(self, arg)
+            arguments
+                self
+                arg.surf (1,1) logical = false
+            end
+
+            for t = 0.5:1.0:50
+                self.net1.advance(t);
+                self.net2.advance(t);
+                self.verifyEqual(self.r1.contents.Y, self.r2.contents.Y, ...
+                                 'RelTol', 5e-4, 'AbsTol', 1e-6);
+                self.verifyEqual(self.r1.T, self.r2.T, 'RelTol', 5e-5);
+                self.verifyEqual(self.r1.P, self.r2.P, 'RelTol', 1e-6);
+
+                if arg.surf
+                    self.verifyEqual(self.interface1.coverages, ...
+                                     self.interface2.coverages, ...
+                                     'RelTol', 1e-4, 'AbsTol', 1e-8);
+                end
+            end
+        end
+
+    end
+
+    methods (Test)
+
+        function testReactorSurfaceType(self)
+            self.makeReactors('addSurf', true);
+            self.verifyTrue(isa(self.surf1, 'ReactorSurface'));
+            self.verifyTrue(startsWith(self.surf1.name, 'ReactorSurface_'));
+            self.surf1.name = 'name-of-reactor-surface';
+            self.verifyEqual(self.surf1.name, 'name-of-reactor-surface');
+        end
+
+        function testClosed(self)
+            self.makeReactors();
+            self.integrate();
```

Comment:
Can you move the assertions out of the `integrate()` function and into a function with a more appropriate name?